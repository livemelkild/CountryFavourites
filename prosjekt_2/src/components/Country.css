@media (prefers-reduced-motion: no-preference){
    #image{
      animation: moveCircle infinite 3s linear;
    }
  }


  @keyframes moveCircle {
      0%{
        transform: rotate(10deg);
      }
      50%{
        transform: rotate(-10deg)
      }
      100%{
        transform: rotate(10deg);
      }
    }
