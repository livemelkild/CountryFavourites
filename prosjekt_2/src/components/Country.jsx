import React from "react"




function Country(props) {
  return(
      <button className = "country" onClick = {props.onClick}>
        <h1>{props.name}</h1>
        <p> {props.description}</p>
        <img src = {props.img} />
      </button>
  )
}

export default Country;
