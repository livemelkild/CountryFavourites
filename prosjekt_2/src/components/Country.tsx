import React, {useState, useEffect} from "react";
import { StringLiteral } from "typescript";

interface PoemI{
  title: string;
  author: string;
  lines: string[];
  linecount: string;
}

interface countryProps{
  name: string;
  description: PoemI;
  sound : string;
  img: string;
  theme: 'light' | 'dark';
  onClick: (event: React.MouseEvent<HTMLDivElement>) => void;
}

let dataLocal: boolean;
let dataSession: boolean;

function Country(props: countryProps) {


  //isFavorite er en variabel som er satt til false, og setIsFavorite er en funksjon som kan brukes hvor jeg vil
  const [isFavorite, setIsFavorite] = useState(false)

  const handleClick = () => {
    setDataLocal(!isFavorite);
    console.log(localStorage.getItem(props.name))
    getData();
    }

  const setDataLocal = (isSaved: boolean) => {
    //set data with localstorage
    localStorage.setItem(props.name, JSON.stringify(isSaved));
  }

  const getData = () => {
    let dataLocal = JSON.parse(localStorage.getItem(props.name) || "false") as boolean;
    console.log(dataLocal);
    setIsFavorite(dataLocal);
    }

  useEffect(() => {
    getData()
  }, [])

  useEffect(()=> {
    console.log(props.
    description);
  }, [props.
  description])

console.log(props.theme)



useEffect(() => {
  getData()
}, [])

  return(
    <div className="content_box">
      <div className = {`country ${props.theme === 'light' ?  '': 'dark'}`}>
          <h1>{props.name}</h1>
          {props.description && (
              <div className = "poem">
              <div className="save_poem" onClick={() => savePoem()}> </div>
              <p><b> {props.description[0].title}</b></p>
              { props.description[0].lines.map((line, i) => (
                <p>{line}</p>
              ))}
              </div>
            )}
          <div id = "sound">
            <audio controls src = {props.sound} />
          </div>

            <div id ="image_style" onClick = {props.onClick}>
            <div id = "image">
              <img src = {process.env.PUBLIC_URL + props.img} alt = "img"/>
            </div>
            </div>
            <br/>
          <div id="like">
          {isFavorite && <div className="saved"  onClick={() => handleClick()} > <img src={process.env.PUBLIC_URL + "./Image/photo_like/ikke_hjerte.png"} alt="hjerte" width="25" height="20"/></div>}
          {!isFavorite && <div className="saved"  onClick={() => handleClick()} > <img src={process.env.PUBLIC_URL + "./Image/photo_like/hjerte.png"} alt="hjerte" width="25" height="20"/></div>}
          </div>
      </div>

    </div>
  )
}

export default Country;
