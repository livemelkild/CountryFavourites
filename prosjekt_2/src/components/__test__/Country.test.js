import React from "react";
import renderer from "react-test-renderer";
import Country from "./../Country";

it("renders without crashing", () => {
    const tree = renderer.create(<Country />).toJSON();
    expect(tree).toMatchSnapshot();
});