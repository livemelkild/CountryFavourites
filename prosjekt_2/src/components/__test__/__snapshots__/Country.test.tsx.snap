// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="country"
>
  <h1 />
  <p>
     
  </p>
  <div
    id="sound"
  >
    <audio
      controls={true}
    />
  </div>
  <div
    id="image_style"
  >
    <div
      id="image"
    >
      <img
        alt="img"
        src="undefined"
      />
    </div>
  </div>
  <br />
  <div
    id="like"
  >
    <div
      className="saved"
      onClick={[Function]}
    >
       
      <img
        alt="hjerte"
        height="20"
        src="./Image/photo_like/hjerte.png"
        width="25"
      />
    </div>
  </div>
</div>
`;
